File: graphicCreator.js
const fs = require("fs");
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.question("Digite o titulo do grafico:  \n", function (name) {
  rl.question(
    "Digite os valores para a criação do grafico: \n",
    function (values) {
      const Values = values.split(" ").sort(function (a, b) {
        return a - b;
      });
      const countedValues = count_duplicate(Values);
      const ParsedKeys = Object.keys(countedValues).map((key) => {
        return `${key},${countedValues[key]},${(
          (countedValues[key] / Values.length) *
          100
        )
          .toPrecision(2)
          .toString()
          .substring(0, 3)}%\n`;
      });
      const Footer = `Total, ${Values.length}, 100%`;
      const Header = `${name}, frequência absoluta (FA), frequência relativa (FR) %\n`;
      const ParsedData = `${Header} ${ParsedKeys} ${Footer}`;
      fs.writeFile(`${name}.csv`, ParsedData, callBackFunction);
      rl.close();
    }
  );
});

function callBackFunction(err) {
  if (err) throw err;
  console.log("Graphic created");
}
function count_duplicate(value) {
  let counts = {};

  for (let i = 0; i < value.length; i++) {
    if (counts[value[i]]) {
      counts[value[i]] += 1;
    } else {
      counts[value[i]] = 1;
    }
  }
  return counts;
}

File: graphicCreator.py
import re
from collections import Counter

def truncate(num):
    return re.sub(r'^(\d+\.\d{,2})\d*$',r'\1',str(num))

graphicName = input('Digite o titulo do grafico: \n')
graphicValues = input('Digite os valores para a criação do grafico: \n')

valuesList = graphicValues.split(" ")
countedValues = Counter(valuesList)

grapihic = open(graphicName + ".csv", "a")

grapihic.write(graphicName + ', frequência absoluta (FA), frequência relativa (FR) %\n')
graphicLine = "{}, {}, {}% \n"

for key in countedValues.keys():
    grapihic.write(
        graphicLine.format(
            key, countedValues[key], truncate((countedValues[key] / len(valuesList)) * 100)
        )
    )
    
graphicFooter = 'Total, {}, 100%'
grapihic.write(graphicFooter.format(len(valuesList)))
grapihic.close()
